app/code/Magento/StoreGraphQl/Controller/HttpRequestValidator/StoreValidator.php

```diff
@@ -44,13 +44,19 @@ public function validate(HttpRequestInterface $request): void
         if (!empty($headerValue)) {
             $storeCode = ltrim(rtrim($headerValue));
             $stores = $this->storeManager->getStores(false, true);
-            if (!isset($stores[$storeCode])) {
-                if (strtolower($storeCode) !== 'default') {
-                    $this->storeManager->setCurrentStore(null);
+            try {
+                if (!isset($stores[$storeCode]) && strtolower($storeCode) !== 'default') {
                     throw new GraphQlInputException(
-                        __("Requested store is not found")
```

Comment:
Please do not use exceptions for conditional logic.
Let's do not expose additional info. And if store is not found or not active `Requested store is not found` should be returned.